src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -689,7 +689,7 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
       (underlyingType(baseType), underlyingType(idxType)) match {
         case (Single(base), Single(idx)) => (base, idx) match {
           case (ArrayT(_, elem), IntT(_)) => elem
-          case (PointerT(ArrayT(_, elem)), IntT(_)) => elem
+          case (ActualPointerT(ArrayT(_, elem)), IntT(_)) => elem
```

Comment:
(repeat) I think a ghost pointer to an array is ok, i.e. this line should account for ghost pointer too.